### PR TITLE
Iframe support

### PR DESCRIPTION
**Pull request by pavelpenchev**

### CODE REVIEW

File: src/main/collectors/BrowserCollector.js [hunk 1]
```diff
@@ -13,13 +17,16 @@ class BrowserCollector {
...
     writer.write({
-      "type" : "browser",
-      "location" : window.location.href,
-      "referrer" : document.referer,
-      "language" : window.navigator.userLanguage || window.navigator.language,
-      "width" : window.screen.width,
-      "height" : window.screen.height,
+      "type" : this.getType(),
+      "location" : win.location.href,
+      "referrer" : doc.referer,
```
Comment (martin-g) [inline src/main/collectors/BrowserCollector.js]: doc.refer`r`er

File: src/main/collectors/InstantSearchQueryCollector.js [hunk 2]
```diff
@@ -27,7 +29,8 @@ class InstantSearchQueryCollector {
    * @param {object} writer - The writer to send the data to
    */
   attach(writer) {
-    var searchBox = document.querySelector(this.searchFieldSelector);
+    var doc = this.getContext() ? this.getContext().getDocument() : document;
```
Comment (martin-g) [inline src/main/collectors/InstantSearchQueryCollector.js]: is it a good idea to introduce helper methods in AbstractCollector:
```
getDocument() {
  return this.getContext() ? this.getContext().getDocument() : document;
} 
```

and similar for `Window`.
This way the specializations can just use `const doc = getDocument();`

File: src/main/collectors/SearchResultCollector.js [hunk 3]
```diff
@@ -13,6 +15,7 @@ class SearchResultCollector {
    * @param {function} countResolver - Function that should return the numnber of results in the search
    */
   constructor(phraseResolver, countResolver) {
+    super("search");
```
Comment (martin-g) [inline src/main/collectors/SearchResultCollector.js]: `InstantSearchQueryCollector` uses the same `type`. Is this OK ?

File: src/main/collectors/InstantSearchQueryCollector.js (hunk 2 above)
Comment (pavelpenchev) [reply src/main/collectors/InstantSearchQueryCollector.js]: Good point, adjusted

File: src/main/collectors/SearchResultCollector.js (hunk 3 above)
Comment (pavelpenchev) [reply src/main/collectors/SearchResultCollector.js]: It's ok, they are both sending search events

**Pull request merged**